test/api/hh_sim.cpp

```diff
@@ -0,0 +1,97 @@
+// NOTE: this assumes neuronapi.h is on your CPLUS_INCLUDE_PATH
+#include "neuronapi.h"
+#include <dlfcn.h>
+
+#include <array>
+#include <cassert>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+
+#include "./test_common.h"
+
+using std::cout;
+using std::endl;
+using std::ofstream;
+
+constexpr std::array<double, 3> EXPECTED_V{
+#ifndef CORENEURON_ENABLED
+    -0x1.04p+6,
+    -0x1.b254ad82e20edp+5,
+    -0x1.24a52af1ab463p+6,
+#else
+    -0x1.04p+6,
+    -0x1.b0c75635b5bdbp+5,
+    -0x1.24a84bedb7246p+6,
+#endif
+};
+
+static const char* argv[] = {"hh_sim", "-nogui", "-nopython", nullptr};
+
+extern "C" void modl_reg(){};
+
+int main(void) {
+    Section* soma;
+    Object* iclamp;
+    Object* v;
+    Object* t;
+    char* temp_str;
+
+    nrn_init(3, argv);
+
+    // load the stdrun library
+    temp_str = strdup("stdrun.hoc");
+    nrn_str_push(&temp_str);
+    nrn_function_call(nrn_symbol("load_file"), 1);
+    nrn_double_pop();
+    free(temp_str);
+
+    // topology
+    soma = nrn_section_new("soma");
+    nrn_nseg_set(soma, 3);
+
+    // define soma morphology with two 3d points
+    nrn_section_push(soma);
+    for (double x: {0, 0, 0, 10}) {
+        nrn_double_push(x);
+    }
+    nrn_function_call(nrn_symbol("pt3dadd"), 4);
+    nrn_double_pop();  // pt3dadd returns a number
+    for (double x: {10, 0, 0, 10}) {
+        nrn_double_push(x);
+    }
+    nrn_function_call(nrn_symbol("pt3dadd"), 4);
+    nrn_double_pop();  // pt3dadd returns a number
+
+    // ion channels
+    nrn_mechanism_insert(soma, nrn_symbol("hh"));
+
+    // current clamp at soma(0.5)
+    nrn_double_push(0.5);
+    iclamp = nrn_object_new(nrn_symbol("IClamp"), 1);
+    nrn_property_set(iclamp, "amp", 0.3);
+    nrn_property_set(iclamp, "del", 1);
+    nrn_property_set(iclamp, "dur", 0.1);
+
+    // setup recording
+    v = nrn_object_new(nrn_symbol("Vector"), 0);
+    nrn_rangevar_push(nrn_symbol("v"), soma, 0.5);
+    nrn_double_push(5.);
```

Comment:
Maybe I would record a bit more frequently, e.g. 1 or 2 msec? applies to other tests as well

Reply:
Someone said that checking the first/middle/last was sufficient :D Also, we have other tests which go to 100ms... Do you want e.g. 10 comparisons? per test

Reply:
hehe.. true!

For the correctness its fine to check the last result I think. To make sure "recording API" itself is working correctly, I thought would be nice to have a bit more value. ~10 sounds reasonable (e.g. per 1msec if tstop is 10)